test/functional/sqlsrv/sqlsrv_ae_fetch_phptypes.phpt

```diff
@@ -0,0 +1,205 @@
+                    "time", "datetimeoffset", "datetime2", "decimal(32,4)", "numeric(36,4)"
+                    );
+
+set_time_limit(0);
+sqlsrv_configure('WarningsReturnAsErrors', 1);
+
+// Connect
+$connectionInfo = array("CharacterSet"=>"UTF-8");
+$conn = AE\connect($connectionInfo);
+if (!$conn) {
+    fatalError("Could not connect.\n");
+}
+
+$tableName = "type_conversion_table";
+$stmt = sqlsrv_query($conn, "IF OBJECT_ID('$tableName', 'U') IS NOT NULL DROP TABLE $tableName");
+$columns = array();
+$insertQuery = "";
+
+FormulateSetupQuery($tableName, $dataTypes, $columns, $insertQuery, $strsize, $strsize2);
+
+$stmt = AE\createTable($conn, $tableName, $columns);
+if (!$stmt) {
+    fatalError("Failed to create table $tableName\n");
+}
+    
+// The data we test against is in values.php
+for ($v = 0; $v < sizeof($values);++$v)
+{
+    // Each value must be inserted twice because the AE and non-AE column are side by side.
+    $testValues = array();
+    for ($i=0; $i<sizeof($values[$v]); ++$i) {
+        $testValues[] = $values[$v][$i];
+        $testValues[] = $values[$v][$i];
+    }
+
+    // Insert the data using sqlsrv_prepare()
+    // Insert one set of data for each PHPTYPE
+    $stmt = sqlsrv_prepare($conn, $insertQuery, $testValues);
+    if ($stmt == false) {
+        print_r(sqlsrv_errors());
+        fatalError("sqlsrv_prepare failed\n");
+    }
+
+    for ($i = 0; $i < sizeof($SQLSRV_PHPTYPE_CONST); ++$i) {
+        if (sqlsrv_execute($stmt) == false) {
+            print_r(sqlsrv_errors());
+            fatalError("sqlsrv_execute failed\n");
+        }
+    }
+
+    $selectQuery = "SELECT * FROM $tableName";
+
+    // Two select statements for selection using 
+    // sqlsrv_get_field and sqlsrv_fetch_array
+    $stmt = sqlsrv_query($conn, $selectQuery);
+    if ($stmt == false) {
+        print_r(sqlsrv_errors());
+        fatalError("First sqlsrv_prepare failed\n");
+    }
+    
+    $stmt2 = sqlsrv_query($conn, $selectQuery);
+    if ($stmt2 == false) {
+        print_r(sqlsrv_errors());
+        fatalError("Second sqlsrv_prepare failed\n");
+    }
+
+    $metadata = sqlsrv_field_metadata($stmt);
+    $numFields = count($metadata);
+
+    $i = 0;
+    $valueAE = null;
+    $valueArrayAE = null;
+    
+    while ($result = sqlsrv_fetch($stmt)) {
+        $dataArray = sqlsrv_fetch_array($stmt2, SQLSRV_FETCH_NUMERIC);
+        
+        for ($j = 0; $j < $numFields; $j++) {
+            $value = sqlsrv_get_field($stmt, $j, $SQLSRV_PHPTYPE_CONST[$i]);
+            $valueArray = $dataArray[$j];
+            
+            // PHPTYPE_STREAM returns a PHP resource, so check the type
+            if (is_resource($value)) $value = get_resource_type($value);
+
+            // For each type, the AE values come first and non-AE values second
+            // So let's do the comparison every second field
+            if ($j%2 == 0) {
+                $valueAE = $value;
+                $valueArrayAE = $valueArray;
```

Comment:
the variable names valueArray and valueArrayAE is a little confusing as it sounds like the variable should be of type array (but actually it is a value retrieved from sqlsrv_fetch_array. Perhaps rename them to valueFromArray and valueFromArrayAE respectively?